### ISSUE TITLE
subdivisionContainer uses deprecated China code

### ISSUE THREAD

**Issue by kagagnon** [punic/punic#158]
- Punic Version 3.2.0

In the auto-generated files, `subdivisionContainment.php` uses the deprecated code for China subdivisions such as `chah` while the data file uses the new code `ch34`. This happens for all subdivision that has a new numeric code.

This issue raise the following error when using `Territory::getList('s')`:

> Undefined index: cnah

**Comment by c960657** [comment]
Just to clarify: You made a typo – the codes in question are `cn34` and `cnah` :-)

This issue was fixed upstream and is included in CLDR 34:
https://unicode.org/cldr/trac/changeset/14433

This should still be fixed in Punic, so we don't throw errors. In case there is string to use, I suggest we create a synthetic fallback string based on the country code, e.g. “China (34)”. This is of course a hack – the proper fix is to upgrade to CLDR 34.

**Comment by mlocati** [comment]
The problem with CLDR 34 is that they removed telephone data in that version, and I still have to think to an alternative.
What about patching the generated json of CLDR 33 before building the PHP data for Punic?

**Comment by c960657** [comment]
I'd prefer copying the telephone data from CLDR 33 into future punic-data releases, rather than patching data from future CLDR releases back into to CLDR 33.

The telephone data is not updated in CLDR anymore, so we could simply remove it from the data repository and just use the existing file in src/data.

I suggest we deprecate the telephone functions in Punic rather than trying to include non-CLDR in the Punic library. The telephony functions could possibly be moved into a separate library based on some other data source, unless such a library already exists.

**Comment by kagagnon** [comment]
Yes, @c960657, you're right, I interchanged the value, `cnah` is the *new* code.

I also agree with you, a library that is used as a wrapper of, or is dependent of another library should strongly use the latest versions of the said library. They removed the telephone data for a good reason.

>[No longer kept up to date, and there are better open-source solutions.](https://unicode.org/cldr/trac/ticket/10383)

Nonetheless, deprecating the telephone data sounds good to me since I guess punic wants to be a close a CLDR as possible.

**Comment by Dazag** [comment]
Hi all! Shouldn't be this issue closed?

**Comment by J2021-ART** [comment]
INDIVIDUAL LOAN OFFER WITHIN 48 HOURS

Hello dear sir or madam
I am a BELGIAN individual who offers international loans. Having a capital which will be used to grant loans between individuals in the short and long term ranging from 2000 € to 500,000 € to all serious people being in the really in need, the interest rate is 2% per year.
I grant financial loans, real estate loans, investment loans,
Car loan, Personal loan, Loan of any kind even if you are registered by the bank or if you are retired
Contact us by
E-mail: inspecteur.joelverhaegheedmond@gmail.com
Phone / WhatsApp: +32 487 83 88 16
I am available to secure your loan application.
cordially
